scripts/gen_quick_start_module.py

```diff
@@ -1,20 +1,152 @@
+DEFAULT = "default"
+ENABLE = "enable"
+DISABLE = "disable"
+
+# Mapping json to release matrix is here for now
+# TBD drive the mapping via:
+#  1. Scanning release matrix and picking 2 latest cuda versions and 1 latest rocm
+#  2. Possibility to override the scanning algorithm with arguments passed from workflow
+
+acc_arch_ver_map = {
+    "nightly": {
+        "accnone": ("cpu", ""),
+        "cuda.x": ("cuda", "11.6"),
+        "cuda.y": ("cuda", "11.7"),
+        "rocm5.x": ("rocm", "5.2")
+        },
+    "release": {
+        "accnone": ("cpu", ""),
+        "cuda.x": ("cuda", "11.6"),
+        "cuda.y": ("cuda", "11.7"),
+        "rocm5.x": ("rocm", "5.2")
+        }
+    }
 
+LIBTORCH_DWNL_INSTR = {
+        PRE_CXX11_ABI: "Download here (Pre-cxx11 ABI):",
+        CXX11_ABI: "Download here (cxx11 ABI):",
+        RELEASE: "Download here (Release version):",
+        DEBUG: "Download here (Debug version):",
+    }
 
 def read_published_versions():
     with open(os.path.join(BASE_DIR, "published_versions.json")) as fp:
         return json.load(fp)
 
+def write_published_versions(versions):
+    with open(os.path.join(BASE_DIR, "published_versions.json"), "w") as outfile:
+            json.dump(versions, outfile, indent=2)
+
+def read_matrix_for_os(osys: OperatingSystem, value: str):
+    try:
+        with open(os.path.join(BASE_DIR, f"{osys.value}_{value}_matrix.json")) as fp:
+            return json.load(fp)["include"]
+    except FileNotFoundError as e:
+        raise ImportError(f"Release matrix not found for: {osys.value} error: {e.strerror}") from e
+
 
 def read_quick_start_module_template():
     with open(os.path.join(BASE_DIR, "_includes", "quick-start-module.js")) as fp:
         return fp.read()
 
+def update_versions(versions, release_matrix, release_version):
+    version_map = {
+        "preview": "preview",
+    }
+    version = ""
+    acc_arch_map = acc_arch_ver_map[release_version]
+
+    if(release_version == "nightly"):
+        version = "preview"
+    else:
+        version = release_matrix[OperatingSystem.LINUX.value][0]["stable_version"]
+
+    # Generating for a specific version
+    if(version != "preview"):
+        version_map = {
+            version: version,
+        }
+        if version not in versions["versions"]:
+            import copy
+            new_version = copy.deepcopy(versions["versions"]["preview"])
+            versions["versions"][version] = new_version
+            versions["latest_stable"] = version
+
+    # Perform update of the json file from release matrix
+    for ver, ver_key in version_map.items():
+        for os_key, os_vers in versions["versions"][ver_key].items():
+            for pkg_key, pkg_vers in os_vers.items():
+                for acc_key, instr in pkg_vers.items():
+
+                    package_type = pkg_key
+                    if pkg_key == 'pip':
+                        package_type = 'manywheel' if os_key == OperatingSystem.LINUX.value else 'wheel'
```

Comment:
Please turn this one into a function (and also, avoid mixing single and double quotes for string literals)